crypto/evp/p_lib.c

```diff
@@ -2052,6 +2054,10 @@ int evp_pkey_copy_downgraded(EVP_PKEY **dest, const EVP_PKEY *src)
         }
     }
 
+    if (allocpkey != NULL) {
```

Comment:
This guard is not really needed.
I suggest removing it, also for defensiveness such that `*dest = NULL` on error.

Reply:
Oops, I had not taken into account that `*dest` may be non-NULL already initially.